### ISSUE TITLE
GBQ: Handle deprecation of oauth2client.tools.run()

### ISSUE THREAD

**Issue by jtratner** [pydata/pandas#8327]
Right now, when you use GbqConnector, you get the following warning:

>WARNING:root:This function, oauth2client.tools.run(), and the use of the gflags library are deprecated and will be removed in a future version of the library.

I'm not sure how quickly this deprecation will happen or if there are alternatives to keep things the way that they are; but at least worthwhile to have an issue here to address it (in the future).

As an aside, I've been glancing at the source for the `bq` client that Google provides as part of Google Cloud SDK and I'm wondering how public the methods are (i.e., if we could adapt some of the code or functions that `bq.py` uses to authenticate and be able to get credentials set up from `gcloud auth login` rather than requiring a separate oauth flow). Not sure though.

Again, cc @jacobschaer , I'd be interested in your opinion on whether this is worth investigating at all. I've been finding `read_gbq` and `to_gbq` quite useful - thanks for putting it together!

**Comment by aginiewicz** [comment]
Any news on this one? I'm packager of pandas on Arch Linux and gflags is only dependency left that isn't compatible with python 3 - and seems won't be in any time soon.

**Comment by jreback** [comment]
@aginiewicz IIRC gbq is not tested / enabled for py3 at all. I am not sure that status in general.

cc @jacobschaer 
cc @azbones

**Comment by azbones** [comment]
Hey @jtratner, @jacobschaer and I are going to try to look at the following in the next week:

-options for Google's deprecating oauth2client.tools.run() method
-that feature that I think someone was adding to allow server auth
-python 3 support (we had some concern when we first considered this, but we'll look into again)

**Comment by azbones** [comment]
@jacobschaer @sean-schaefer @jtratner  Also, I just saw https://github.com/pydata/pandas/issues/9713, so we can look at testing again as we should catch things like that.

**Comment by cgrin** [comment]
@azbones Just came over to take a look at this and did a bit of digging. It's actually not a huge change, and I've already written a patch to fix it. The replacement for [tools.run](https://google-api-python-client.googlecode.com/hg/docs/epy/oauth2client.old_run-module.html) is [tools.run_flow()](https://google-api-python-client.googlecode.com/hg/docs/epy/oauth2client.tools-module.html). Change the import from 
```python
from oauth2client.tools import run
```
to 

```python
from oauth2client.tools import run_flow, argparser
```
and invoke with 
```python
credentials = run_flow(flow, storage, argparser.parse_args([]))
```

Not sure if I should push this to #9722 or wait for that to accept, and make a new pull for this issue.

@jreback @jacobschaer - Thoughts on how to handle this?

**Comment by azbones** [comment]
@cgrin Cool- that is the same conclusion we had come to! We'll leave it to you, but let us know if you need any help.

The other feature we were thinking about in gbq was https://github.com/pydata/pandas/issues/8489 which is generally focused on an improved auth flow for servers. We haven't researched that one yet, but I think someone had taken a cut at it. 

cc: @jacobschaer @sean-schaefer

**Comment by jreback** [comment]
closed by #9722
